plugins/decor/deco-layout.cpp

```diff
@@ -230,7 +243,17 @@ decoration_layout_t::action_response_t decoration_layout_t::handle_press_event(
         auto area = find_area_at(current_input);
         if (area && (area->get_type() & DECORATION_AREA_MOVE_BIT))
         {
-            return {DECORATION_ACTION_MOVE, 0};
+            if (timer.is_connected())
+            {
+                double_click_at_release = true;
+            } else
+            {
+                timer.set_timeout(300, [=] ()
+                {
+                    timer.disconnect();
```

Comment:
So you say that without this disconnect Wayfire crashes?

Reply:
It crashes not, but stays connected: every isolated click after the first one then toggles maximization.

Reply:
The timer is supposed to automatically disconnect if the handler returns false. There was a bug in core - can you try #1086 and without the  manual disconnect here?

Reply:
yes, tested (on 0.7.0) works fine. Thanks!